### ISSUE TITLE
Image rotation?

### ISSUE THREAD

**Issue by den1k** [ethanselzer/react-image-magnify#26]
Is there any way to rotate the image? I had a few attempts in CSS transform rotate and the problem was that the magnifier box always rotated with the image.

**Comment by ethanselzer** [comment]
Hi @den1k - Unfortunately, image rotation is not a supported feature of react-image-magnify. Sorry about that. I think it is possible to rotate the initial image using the `imageClassName` prop, but the enlarged image uses transform internally and does not allow for augmenting the transformation. I think the only way to rotate the enlarged image would be to rotate its container using `enlargedImageContainerClassName` or `enlargedImageContainerStyle`. Note that unless your images are square, rotation may interfere with the position of the lens and enlarged image elements.

**Comment by den1k** [comment]
Yes, I tried it all. And the only reason I made the issue is that after 2h
of trying I couldn't get to a satisfactory result. The viewer would either
rotate with the image or "not understand" that the image has been rotated
and then the lens area didn't represent the enlarged image. I don't mean to
add rotation as a UI element, but it would be nice to allow to let us
rotate the image programmatically. At least in my experience, this is a
very common use case. Thanks for this lib!

**Comment by ethanselzer** [comment]
@den1k - Thanks for the information. Sorry you were not able to achieve the results you want. I'd like to better understand your user case for react-image-magnify. What type of application or feature are you building with it?

**Comment by den1k** [comment]
It's a grid view of user-submitted images. Therefore the rotation is sometimes incorrect.

**Comment by ethanselzer** [comment]
@den1k - Thanks for the explanation. It's helpful for me to understand how the project is being used. I think the focus of react-image-magnify is shopping site detail pages, where image rotation may not be a high priority. I'll put support for image rotation in the backlog, but I'm not sure right now where it fits in the overall strategy of the project.

**Comment by SpringsTea** [comment]
I'm also trying to put rotation into this. I'm achieving this just with css:

`style:{transform: 'rotate(${rotate}deg)'}`

Where I increment `rotate` when the image is clicked. Works well, both the small and large images rotate as expected. The only probably is the hover controls don't keep up. So hovering your cursor downwords will pan the large image upwards if the image is upside down. 

All it would really need is an option to change which axis get moved on hover.
Maybe there would also be another use for an option like that.

**Comment by jothi-pyt** [comment]
> It's a grid view of user-submitted images. Therefore the rotation is sometimes incorrect.

I have a similar use-case which requires having an image rotation. Is there any update on this?
